src/wallet/wallet.h

```diff
@@ -1220,6 +1224,8 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
 
     /** Add a KeyOriginInfo to the wallet */
     bool AddKeyOrigin(const CPubKey& pubkey, const KeyOriginInfo& info);
+
+    friend struct WalletTestingSetup;
```

Comment:
It's needed to avoid "error: cannot cast 'CWallet' to its private base class 'interfaces::Chain::Notifications'" in the WalletTestingSetup constructor:

https://github.com/bitcoin/bitcoin/blob/0ca5446db3cda7f911e911a094482311f6a71221/src/wallet/test/wallet_test_fixture.cpp#L16

because the relevant base class is private:

https://github.com/bitcoin/bitcoin/blob/0ca5446db3cda7f911e911a094482311f6a71221/src/wallet/wallet.h#L641